<launch>
    <arg name="uav_id" default="1"/>
    <!-- 无人机初始位置 -->
	<arg name="uav_init_x" default="0.0"/>
    <arg name="uav_init_y" default="0.0"/>
	<!-- 1.optitrack 2.t265 -->
	<arg name="input_source" default="2"/>
    <group ns="/uav$(arg uav_id)">
        <node pkg="mavros" type="mavros_node" name="mavros" output="screen">
        <param name="fcu_url" value="/dev/ttyACM0:57600" />
        <!--param name="gcs_url" value="udp://@192.168.31.46" / -->
        <param name="gcs_url" value="udp://@192.168.31.96" />
        <param name="target_system_id" value="1" />
        <param name="target_component_id" value="1" />
        <rosparam command="load" file="$(find command)/config/px4_pluginlists.yaml" />
        <rosparam command="load" file="$(find command)/config/px4_config.yaml" />
    </node>
    <include file="$(find command)/launch_basic/rs_d400_and_t265.launch">
    </include> 
    </group> 

	<!-- publish tf -->
	 <include file="$(find command)/launch_basic/cam_tf.launch">
    	</include>
	<!-- publish cam pose -->
	<node pkg="command" name="pub_campose" type="pub_campose.py"> 
	</node>
    <node pkg="command" name="uav$(arg uav_id)_vision2px4" type="vision_estimator" output="screen">
        <param name="uav_id" value="$(arg uav_id)"/>
        <param name="input_source" value="$(arg input_source)"/>
        <param name="uav_init_x" value="$(arg uav_init_x)" type="double"/>
        <param name="uav_init_y" value="$(arg uav_init_y)" type="double"/>
    </node>
    
   
</launch>